### PR TITLE
Interoperable ra tls support for librats

### PR DESCRIPTION
**Pull request by KB5201314**
This submission ports the support for Interoperable RA-TLS X.509 Cert from rats-tls to librats. Specifically, this submission:
- Introduces two new APIs: librats_get_attestation_certificate() and librats_verify_attestation_certificate().
- Adds rats-tls's crypto_wrappers as a dependency to librats, which is necessary for implementing Interoperable RA-TLS X.509 Cert. Two external dependency libraries, SGXSSL and libcbor, are also introduced.
- Porting the following PRs to librats:
  - https://github.com/inclavare-containers/rats-tls/pull/123
  - https://github.com/inclavare-containers/rats-tls/pull/135
  - https://github.com/inclavare-containers/rats-tls/pull/144
  - https://github.com/inclavare-containers/rats-tls/pull/147
  - https://github.com/inclavare-containers/rats-tls/pull/150
  - https://github.com/inclavare-containers/rats-tls/pull/151

### CODE REVIEW

**Review by zeuson0 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -33,23 +33,31 @@
 
  2. Build and install librats
 
+ ```shell
+ cd rats-tls
+ cmake -DBUILD_SAMPLES=on -H. -Bbuild
+ make -C build install
+ ```
+
+ `cert-app` will be installed to `/usr/share/librats/samples/cert-app` on your system. All instances are placed in `/opt/rats-tls/lib`.
```
Comment (haosanzi) [inline README.md:42]: please change `rats-tls` to `librats` .  I remember the path is `/usr/local/lib/librats/`, please check it. Thank you

File: samples/cert-app/common.c [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				size_t *certificate_size_out)
+{
+	uint8_t *private_key = NULL;
+	size_t private_key_size = 0;
+
+	int ret = -1;
+	rats_attester_err_t rats_ret;
+
+	if (no_privkey) {
+		printf("The flag no_privkey is true. We will let librats to generate random key pairs.\n");
```
Comment (haosanzi) [inline samples/cert-app/common.c:144]: Could you please use RATS_xx rather than print?

File: samples/cert-app/main.c [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+			debug_enclave = true;
+			break;
+		case 'k':
+			no_privkey = true;
+			break;
+		case 'c':;
+			const char *divider = strchr(optarg, ':');
+			if (!divider) {
+				printf("Invalid argment '%s', shall in format: 'key:val'\n",
+				       optarg);
```
Comment (haosanzi) [inline samples/cert-app/main.c:226]: Did you test this code use nullattester + nullverifier + opensslcrypto?

File: CMakeLists.txt [hunk 4]
```diff
@@ -250,21 +276,23 @@ elseif(WASM)
...
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/attester.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/verifier.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/core.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/claim.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/err.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/log.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/conf.h
-                       ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/sgx.h
                        ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/evidence.h
+                       ${CMAKE_CURRENT_SOURCE_DIR}/include/librats/hash.h
```
Comment (zeuson0) [inline CMakeLists.txt:295]: Is `${CMAKE_CURRENT_SOURCE_DIR}/include/librats/endorsement.h` missing here?

**Review by haosanzi — APPROVED** [review]

**Pull request merged**